processing/src/main/java/org/apache/carbondata/processing/merger/CompactionResultSortProcessor.java

```diff
@@ -190,7 +200,7 @@ public boolean execute(List<RawResultIterator> unsortedResultIteratorList,
       LOGGER.error(e.getLocalizedMessage(), e);
       throw e;
     } finally {
-      if (partitionSpec != null) {
```

Comment:
Instead of passing boolean value, cant we directly get `CarbonCommonConstants.CARBON_MERGE_INDEX_IN_SEGMENT `from properties? and the if check should be - if merge index not enabled, then write segment file. As below, 

     ```  boolean isMergeIndexEnable = Boolean.parseBoolean(CarbonProperties.getInstance().getProperty(
            CarbonCommonConstants.CARBON_MERGE_INDEX_IN_SEGMENT,
            CarbonCommonConstants.CARBON_MERGE_INDEX_IN_SEGMENT_DEFAULT));
        if (partitionSpec != null && !isMergeIndexEnable) {  ```

Reply:
yes, @liuhe0702 instead of making parameter changes or adding constructors, check property here only and can avoid changes in MergerRDD

Reply:
done